File: 프로그래머스/2/12941. 최솟값 만들기/최솟값 만들기.py
def solution(A, B):
    answer = 0

    # A는 오름차순, B는 내림차순 정렬
    A.sort()
    B.sort(reverse=True)

    # 같은 인덱스끼리 곱해서 누적
    for i in range(len(A)):
        answer += A[i] * B[i]

    return answer

File: 프로그래머스/2/12941. 최솟값 만들기/최솟값 만들기.js
function solution(A, B) {
    let answer = 0;

    // A 오름차순 정렬
    A.sort((a, b) => a - b);

    // B 내림차순 정렬
    B.sort((a, b) => b - a);

    // 같은 인덱스끼리 곱해서 누적
    for (let i = 0; i < A.length; i++) {
        answer += A[i] * B[i];
    }

    return answer;
}
